### ISSUE TITLE
即使在训练集上，预测效果也非常差

### ISSUE THREAD

**Issue by ljch2018** [hiyouga/ChatGLM-Efficient-Tuning#191]
1、训练集loss非常低：
 83%|████████▎ | 200/240 [27:39<05:26,  8.16s/it]{'loss': 0.0011, 'learning_rate': 5.367076727981382e-05, 'epoch': 15.05}

2、预测代码：
python src/train_sft.py \
        --do_predict \
        --model_name_or_path glm/chatglm2-6b \
        --use_v2 \
        --dataset  dataset-cs \
        --max_source_length 3000 \
        --max_target_length 100 \
        --checkpoint_dir sft/checkpoint-200 \
        --output_dir pred \
        --per_device_eval_batch_size 2 \
        --predict_with_generate \
        --fp16

3、预测结果
<img width="1375" alt="image" src="https://github.com/hiyouga/ChatGLM-Efficient-Tuning/assets/22562546/eaa060d5-562c-4c8e-ab87-5dd5294b4582">

**Comment by hiyouga** [comment]
有数据集吗？我本地复现一下

**Comment by ljch2018** [comment]
> 有数据集吗？我本地复现一下

内部私密数据，不方便发出现。

**Comment by hiyouga** [comment]
之前的预测脚本有 bug，现在已经更新，重新跑一下

**Comment by LiZhangMing** [comment]
> 1、训练集loss非常低： 83%|████████▎ | 200/240 [27:39<05:26, 8.16s/it]{'loss': 0.0011, 'learning_rate': 5.367076727981382e-05, 'epoch': 15.05}
> 
> 2、预测代码： python src/train_sft.py --do_predict --model_name_or_path glm/chatglm2-6b --use_v2 --dataset dataset-cs --max_source_length 3000 --max_target_length 100 --checkpoint_dir sft/checkpoint-200 --output_dir pred --per_device_eval_batch_size 2 --predict_with_generate --fp16
> 
> 3、预测结果 <img alt="image" width="1375" src="https://user-images.githubusercontent.com/22562546/249086667-eaa060d5-562c-4c8e-ab87-5dd5294b4582.png">

现在可以了吗你这边
